File: Project/Data/Models/MenuItem.cs
using Project.Data.Enums;

namespace Project.Data.Models
{
    public class MenuItem
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public int CategoryId { get; set; }
        public Category Category { get; set; } = null!;
        public ItemType ItemType { get; set; } = ItemType.Drink;
        public bool IsAlcoholic { get; set; }
        public double? AlcoholPercentage { get; set; }
        public string? ImageUrl { get; set; }
        public bool IsAvailable { get; set; } = true;
        public ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public string? UpdatedBy { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: frontend/src/types/menuItem.ts
export type ItemType = 'Drink' | 'Food';

export interface MenuItemDto {
  id: number;
  name: string;
  price: number;
  categoryId: number;
  categoryName: string;
  itemType: ItemType;
  isAlcoholic: boolean;
  alcoholPercentage?: number;
  imageUrl?: string;
  isAvailable: boolean;
  createdBy: string;
  createdAt: string;
  updatedBy?: string;
  updatedAt?: string;
}

export interface CreateMenuItemDto {
  name: string;
  price: number;
  categoryId: number;
  itemType: ItemType;
  isAlcoholic: boolean;
  alcoholPercentage?: number;
  imageUrl?: string;
  isAvailable: boolean;
}
